src/Util/Strings.php

```diff
@@ -0,0 +1,408 @@
+                    // we might be inside a bbcode color tag - leave it alone
+                    continue;
+                }
+
+                if (substr($match, -1, 1) === '.') {
+                    $ret[] = substr($match, 0, -1);
+                } else {
+                    $ret[] = $match;
+                }
+            }
+        }
+
+        // Otherwise pull out single word tags. These can be @nickname, @first_last
+        // and #hash tags.
+
+        if (preg_match_all('/([!#@][^\^ \x0D\x0A,;:?]+)([ \x0D\x0A,;:?]|$)/', $string, $matches)) {
+            foreach ($matches[1] as $match) {
+                if (strstr($match, ']')) {
+                    // we might be inside a bbcode color tag - leave it alone
+                    continue;
+                }
+                if (substr($match, -1, 1) === '.') {
+                    $match = substr($match,0,-1);
+                }
+                // ignore strictly numeric tags like #1
+                if ((strpos($match, '#') === 0) && ctype_digit(substr($match, 1))) {
+                    continue;
+                }
+                // try not to catch url fragments
+                if (strpos($string, $match) && preg_match('/[a-zA-z0-9\/]/', substr($string, strpos($string, $match) - 1, 1))) {
+                    continue;
+                }
+                $ret[] = $match;
+            }
+        }
+
+        return $ret;
+    }
+
+    /**
+     * @brief Check for a valid email string
+     *
+     * @param string $email_address Email address to be evaluated.
+     * 
+     * @return boolean  Value indicating whether or not the string is a valid email address.
+     */
+    public static function isValidEmail($email_address)
+    {
+        return preg_match('/^[_a-zA-Z0-9\-\+]+(\.[_a-zA-Z0-9\-\+]+)*@[a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)+$/', $email_address);
+    }
+
+    /**
+     * @brief Normalize url
+     *
+     * @param string $url   URL to be normalized.
+     * 
+     * @return string   Normalized URL.
+     */
+    public static function normaliseLink($url)
+    {
+        $ret = str_replace(['https:', '//www.'], ['http:', '//'], $url);
+        return rtrim($ret, '/');
+    }
+
+    /**
+     * @brief Normalize OpenID identity
+     * 
+     * @param string $s OpenID Identity
+     * 
+     * @return string   normalized OpenId Identity
+     */
+    function normaliseOpenID($s)
+    {
+        return trim(str_replace(['http://', 'https://'], ['', ''], $s), '/');
+    }
+
+    /**
+     * @brief Compare two URLs to see if they are the same, but ignore
+     * slight but hopefully insignificant differences such as if one
+     * is https and the other isn't, or if one is www.something and
+     * the other isn't - and also ignore case differences.
+     *
+     * @param string $a first url
+     * @param string $b second url
+     * @return boolean True if the URLs match, otherwise False
+     *
+     */
+    public static function compareLink($a, $b)
```

Comment:
Maybe we move it to  `Friendica\Util\URL` class with name `compare($urlA, $urlB)`.
So this class would get a little bit smaller :-)